### ISSUE TITLE
Why use real labels for the fake images?

### ISSUE THREAD

**Issue by wscnwps** [voletiv/self-attention-GAN-pytorch#4]
Why use real labels for the fake images in conditional batchNorm?
It seems that "z" controls the generated sample, and the "label" can only affect the sample by scaling.
https://github.com/voletiv/self-attention-GAN-pytorch/trainer.py
146  # Generate fake images for same real labels
147                      fake_images = self.G(z, real_labels)

The TF code seems to generate fake labels with random?
https://github.com/brain-research/self-attention-gan/model.py
  220     gen_class_logits = tf.zeros((batch_size, self.num_classes))                  
  221     gen_class_ints = tf.multinomial(gen_class_logits, 1)                         
  222     # gen_sparse_class = tf.argmax(gen_class_ints, axis=1)  BIG BUG!!!!!         
  223     gen_sparse_class = tf.squeeze(gen_class_ints)                                
  224     print('han gen_sparse_class.shape', gen_sparse_class.shape)                  
  225     assert len(gen_class_ints.get_shape()) == 2                                  
  226     gen_class_ints = tf.squeeze(gen_class_ints)                                  
' 227     assert len(gen_class_ints.get_shape()) == 1                                  
  228     gen_class_vector = tf.one_hot(gen_class_ints, self.num_classes)              
  229     assert len(gen_class_vector.get_shape()) == 2                                
  230     assert gen_class_vector.dtype == tf.float32

**Comment by voletiv** [comment]
I didn't think it would make too much of a difference to generate fake labels or to just repeat the real labels at that point.
